_playwright-tests/tests/API/Repositories.spec.ts

```diff
@@ -590,4 +594,80 @@ test.describe('Repositories', () => {
       }
     });
   });
+
+  test('Manually trigger repository snapshot', async ({ client, cleanup }) => {
+    const repoName = `manual-snapshot-${randomName()}`;
+    const repoUrl = randomUrl();
+
+    await cleanup.runAndAdd(() => cleanupRepositories(client, repoName, repoUrl));
+
+    let repo: ApiRepositoryResponse;
+    await test.step('Create a repo with snapshot enabled', async () => {
+      repo = await new RepositoriesApi(client).createRepository({
+        apiRepositoryRequest: {
+          name: repoName,
+          url: repoUrl,
+          snapshot: true,
+        },
+      });
+      expect(repo.name).toBe(repoName);
+      expect(repo.url).toBe(repoUrl);
+      expect(repo.snapshot).toBe(true);
+    });
+
+    let FirstRepo: ApiRepositoryResponse;
+    await test.step('Wait for repository to be valid and get initial snapshot info', async () => {
+      const getRepository = () =>
+        new RepositoriesApi(client).getRepository(<GetRepositoryRequest>{
+          uuid: repo.uuid?.toString(),
+        });
+      const waitWhilePending = (resp: ApiRepositoryResponse) => resp.status === 'Pending';
+      const resp = await poll(getRepository, waitWhilePending, 10);
+      expect(resp.status).toBe('Valid');
+      repo = resp;
+
+      FirstRepo = await new RepositoriesApi(client).getRepository(<GetRepositoryRequest>{
```

Comment:
you can grab the repo's lastSnapshotTaskUuid without fetching the repo again since you're saving the response to the repo once it's valid, so you can just do this:

`firstSnapshotTaskUuid = repo.lastSnapshotTaskUuid;`

then compare that string instead of `FirstRepo.lastSnapshotTaskUuid`